clusterfuzz/commands/reproduce.py

```
@@ -12,7 +12,55 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
+import urllib
+import webbrowser
+import urlfetch
+
+
+def get_testcase_info(testcase_id, auth):
+  response = urlfetch.fetch(
+      url='https://staging-dot-cluster-fuzz.appspot.com/v2/testcase-detail/oauth?testcaseId=%s' % testcase_id,
+      headers={'Authorization': auth})
+
+  return response.status, json.loads(response.body), response.headers
+
+
+def generate_oauth_url():
+  client_id = '981641712411-sj50drhontt4m3gjc3hordjmpc7bn50f.apps.googleusercontent.com'
+  scope = 'email profile'
+  redirect_uri = 'urn:ietf:wg:oauth:2.0:oob'
+
+  url = 'https://accounts.google.com/o/oauth2/v2/auth?%s' % (
+      urllib.urlencode({
+          'scope': scope,
+          'client_id': client_id,
+          'response_type': 'code',
+          'redirect_uri': redirect_uri}))
+  return url
+
 
 def execute(testcase_id, current):
   """Execute the reproduce command."""
+
   print 'Reproduce %s (current=%s)' % (testcase_id, current)
+
+  webbrowser.open(generate_oauth_url(), new=1, autoraise=True)
+  code = raw_input('Please enter your verification code:')
+
+  print 'Downloading testcase information...'
+
+  status, body, headers = get_testcase_info(
+      testcase_id,
+      'VerificationCode %s' % code)
+
+  if status != 200:
+    print 'Incorrect or expired verification code'
```

Let's raise an exception to exit instead of `return`.